tfx/components/transform/executor.py

```diff
@@ -329,10 +329,20 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
         raise ValueError('analyze cannot be empty when splits_config is set.')
     else:
       splits_config.analyze.append('train')
-      splits_config.transform.extend(
-          artifact_utils.decode_split_names(
-              artifact_utils.get_single_instance(
-                  input_dict[EXAMPLES_KEY]).split_names))
+      split_names = artifact_utils.decode_split_names(
+          input_dict[EXAMPLES_KEY][0].split_names)
```

Comment:
Should this be using GetValues? (similarly below)
Also, accessing the first element here seems a bit arbitrary, should we assert that it's non-empty and add a comment about the splits being the same for all? (move the comment from below up)